### DISCUSSION TITLE
Heads-up: govc and vcsim artefacts change coming

### DISCUSSION THREAD

**Discussion by embano1** [Homebrew/discussions#1262] [Everyday usage]
Hi,

I'm a maintainer of the vmware/govmomi repository and we're about to change the way artefacts are built and named in the upcoming release (`v0.25.0`, pending). This might affect the homebrew-core formula thankfully provided by the HomeBrew community (see example below). We can optionally provide our own tap or continue to use the the core formula, whatever you and the community prefers and works best. Happy to discuss. More details on the upcoming change in this PR: https://github.com/vmware/govmomi/pull/2340

cc/ @chenrui333

Example: https://github.com/Homebrew/homebrew-core/pull/73383

**Comment by SMillerDev** [comment]
Just make a pull request to update the version that also switches the artifact building. https://docs.brew.sh/Updating-Software-in-Homebrew

**Reply by embano1** [reply]
Thx, just wasn't sure if it would break the action `action-homebrew-bump-formula: Created with brew bump-formula-pr` which seems to be at play here too. 

I am happy to file a PR if it's just these changes:

```ruby
  url "https://github.com/vmware/govmomi/archive/v0.24.1.tar.gz"
  sha256 "cf1bef8b5a44298d724753ee2f0ba443375b3e3b1572c1c7fff8c5775b803e86"
```

**Reply by chenrui333** [reply]
Feel free to drop a PR. :)

**Reply by embano1** [reply]
@chenrui333 now that we have the official `v0.25.0` [release](https://github.com/vmware/govmomi/releases) where we also ship artifacts for govc/vcsim for different architectures, I am happy to file a PR to update the formula.

I am just not an expert on authoring Brew formulas and would appreciate your help here as the current formula builds from source without injecting build time variables, version information, etc. as we do for the releases.

Each [release](https://github.com/vmware/govmomi/releases) now includes artifacts for different architectures including checksums. Can you point me to a working formula which uses pre-built binaries for different architectures (arm, intel) so I can adapt it for govc and file a PR? I'm also not sure how the checksums are calculated and if changes to `action-homebrew-bump-formula` are needed?

Appreciate your feedback!

**Reply by carlocab** [reply]
We don't have any formulae that install [pre-built binaries](https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae). Unless you need a pre-built binary to bootstrap?

**Reply by embano1** [reply]
Oh, understood. So the formula is always expected to build from source then? In this case we'd have to somehow inject the build variables, which is currently done this way for `govc`: https://github.com/vmware/govmomi/blob/master/.goreleaser.yml#L27

**Reply by embano1** [reply]
This is required in order to give users detailed version information, e.g. used for troubleshooting:

```console
$ govc version -l
Build Version: 0.25.0
Build Commit: bb0307eb
Build Date: 2021-04-16T16:00:59Z
```

**Reply by carlocab** [reply]
Sure, you can pass that via `ldflags` in `std_go_args`. Just `grep` for `std_go_args\(ldflags ".*"\)`.

**Reply by embano1** [reply]
Thx. Are there also helper/convenience methods in Brew (formulas) to easily retrieve commit (SHAs), tags, etc. associated with a repo? Or should those be manually defined in the `.rb` file and referenced in all steps, incl. `url`, etc.?

**Reply by embano1** [reply]
I could use something like this

```ruby
def install
    ldflags = "-s -w -X main.version=#{version}"
    system "go", "build", "-ldflags", ldflags, "-o",
```

But this would only give me the `{version}`. `BuildDate` could be generated with a Ruby helper method I guess, but `COMMIT` is tricky as it's not available in this case...

**Reply by SMillerDev** [reply]
If you tagged something it can only ever be a single commit since the [git manual](https://git-scm.com/docs/git-tag#_on_re_tagging) says re-tagging is "the insane thing" to do. So maybe you could just leave that one as an empty string?

**Reply by embano1** [reply]
Yeah, will leave it empty then. So looks like all I need for the PR is the LDFLAGS stuff and not touch any of the checksums etc in the current file?

**Reply by SMillerDev** [reply]
If the version homebrew ships is already the latest, yes.
